local-tests/setup/tinny-utils.ts

```diff
@@ -45,3 +45,21 @@ export function randomSolanaPrivateKey() {
   }
   return result;
 }
+
+/**
+ * Wraps a promise with a timeout.
+ * If the promise does not resolve or reject within the specified time, it will be rejected with a "Timed out" error.
+ *
+ * @param promise - The promise to wrap with a timeout.
+ * @param ms - The timeout duration in milliseconds.
+ * @returns A new promise that resolves or rejects based on the original promise or the timeout.
+ */
+export function withTimeout<T>(
+  promise: Promise<T>,
+  ms: number
+): Promise<T | void> {
```

Comment:
No i see what you mean, since it rejects it wont effect the return value of the `Promise` as the `timeout` promise will not return a value hence `never` but may `throw` and exception. I assumed `never` was assignable to `void` but it is actually a distinct type which is the omission of any return value. 

Going to resolve the previous comment. Thanks for the breakdown.